src/argilla/server/schemas/v1/datasets.py

```diff
@@ -524,6 +524,13 @@ def check_user_id_is_unique(cls, values: Optional[List[UserResponseCreate]]) ->
 
         return values
 
+    @validator("metadata", pre=True)
+    def skip_nan_values(cls, metadata: Optional[Dict[str, Any]]) -> Optional[Dict[str, Any]]:
+        if metadata is None:
+            return metadata
+
+        return {k: v for k, v in metadata.items() if v == v}  # By definition, NaN != NaN
```

Comment:
It is possible to use `math.isnan(value)` to do the check more explicit than `v == v` but I'm not sure if it's really useful because the function raises a `TypeError` exception if it's not a number and here you're checking for other types as well.

Reply:
`math.isnan(value)` will fail with string types and `None`. So we need to add extra conditions.